components/Auth/ForgotPassword.tsx

```diff
@@ -0,0 +1,105 @@
+import { ArrowBackIcon } from '@chakra-ui/icons'
+import { Button, HStack, Input, Text, useToast, VStack } from '@chakra-ui/react'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useHookFormProps } from 'hooks'
+import { AuthState } from 'models/users'
+import React, { Dispatch, SetStateAction } from 'react'
+import { useForm } from 'react-hook-form'
+import { resetPassword } from 'utils/firebase'
+import * as yup from 'yup'
+
+interface ForgotPasswordProps {
+  setCurrentState: Dispatch<SetStateAction<AuthState>>
+}
+
+interface ForgotPasswordForm {
+  email: string
+}
+
+const loginFormSchema = yup.object({
+  email: yup.string().required().email(),
+})
+
+const initialState = {
+  email: '',
+}
+
+const ForgotPassword = ({
+  setCurrentState,
+}: ForgotPasswordProps): JSX.Element => {
+  const toast = useToast()
+
+  const form = useForm({
+    defaultValues: initialState,
+    resolver: yupResolver(loginFormSchema),
+  })
+
+  const onSubmit = async ({ email }: ForgotPasswordForm) => {
+    const response = await resetPassword(email)
+
+    if (response.status === 200) {
+      toast({
+        title: 'Password Reset Email Sent',
+        description: response.message,
```

Comment:
Just curious, do we know if the message we get back here will be user friendly?

Reply:
It is pretty readable but I could also write some custom messages

Reply:
If it's readable as is, no need to write anything custom I think